common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/ClientCertAuthChallengeHandler.java

```diff
@@ -214,168 +107,92 @@ public void run() {
     public Void processChallenge(@NonNull final ClientCertRequest request) {
         //final String methodTag = TAG + ":processChallenge";
         //If a smartcard device is connected, proceed with smartcard CBA.
-        synchronized (sDeviceLock) {
-            if (mDevice != null) {
-                handleSmartcardCertAuth(request);
-                return null;
-            }
+        if (mSmartcardCertBasedAuthManager.isDeviceConnected()) {
+            handleSmartcardCertAuth(request);
+            return null;
         }
         //Else, proceed with user certificates stored on device.
         handleOnDeviceCertAuth(request);
         return null;
     }
 
     /**
-     * Collects YubiKitCertDetails from PIV certificates on YubiKey.
-     * If certificates are found on YubiKey, the method proceeds with the smartcard certificate picker.
+     * Collects ICertDetails from PIV certificates on smartcard device.
+     * If certificates are found on smartcard, the method proceeds with the smartcard certificate picker.
      * Otherwise, appropriate error dialogs are shown.
      * @param request ClientCertRequest received from AzureActiveDirectoryWebViewClient.onReceivedClientCertRequest.
      */
     @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
     private void handleSmartcardCertAuth(@NonNull final ClientCertRequest request) {
         final String methodTag = TAG + ":handleSmartcardCertAuth";
-        //A connection to the YubiKey needs to be made in order to read the certificates off it.
-        getActivePivSessionAsync(request, new IPivSessionCallback() {
+
+        mSmartcardCertBasedAuthManager.attemptDeviceSession(new ISmartcardCertBasedAuthManager.ISessionCallback() {
```

Comment:
This is a bit of a subjective style-nit, so take with a pinch of salt.

`attemptDeviceSession` sounds like the name of method that doesn't think it will succeed..... and maybe that's true, but the name doesn't inspire confidence in the caller.

When these kinds of "maybe this will work!" methods crop-up, I prefer to make the compiler do the talking.

For instance, if the method `throws`, the caller must catch the `Exception` thereby signaling that it can fail.

Alternatively, if the method accepts an `interface` with an error callback, we also know that it can fail.

Reply:
Makes sense... renaming to `requestDeviceSession`.